crates/provider-wit-bindgen-macro/src/lib.rs

```diff
@@ -554,19 +589,19 @@ pub fn generate(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
         /// as performed by the host runtime
         ///
         /// Interfaces imported by the provider can use this to send traffic across the lattice
-        pub struct InvocationHandler<'a> {
-            ld: &'a ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::InterfaceLinkDefinition,
+        pub struct InvocationHandler {
+            target: ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::core::ComponentId,
             wrpc_client: ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::core::wrpc::Client
         }
 
-        impl<'a> InvocationHandler<'a> {
-            pub fn new(ld: &'a ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::core::InterfaceLinkDefinition) -> Self {
+        impl InvocationHandler {
+            pub fn new(target: ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::core::ComponentId) -> Self {
                 let connection = ::wasmcloud_provider_wit_bindgen::deps::wasmcloud_provider_sdk::provider_main::get_connection();
                 // NOTE: The link definition that is used here is likely (source_id=?, target=<provider>)
                 //
                 // todo(vados-cosmonic): This invocation handler should arguably create a uni-directional
                 // "return" link to anything it wants to call, since links are uni-directional now.
-                Self { ld, wrpc_client: connection.get_wrpc_client(&ld.source_id) }
+                Self { target: target.clone(), wrpc_client: connection.get_wrpc_client(target) }
```

Comment:
Sanity check me here, this client already includes the lattice name in there as well correct?

Reply:
Yup, it does! the `target` probably doesn't have to be saved though, so I might be able to remove that